src/tainting/Taint_lval_env.ml

```diff
@@ -79,6 +79,8 @@ let normalize_lval lval =
   let* base, rev_offset =
     match base with
     | Var _ -> Some (base, rev_offset)
+    (* explicit dereference of `this` e.g. `this->x` *)
```

Comment:
You should be able to add an intra-procedural test here. Something like this:

```cpp
ptr->a = "tainted"
ptr->b = "safe"
// ruleid:
sink(ptr->a); 
// ok:
sink(ptr->b);
```

Reply:
Added